pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -54,10 +58,14 @@ import (
 type Repair struct {
 	interval      time.Duration
 	serviceClient corev1client.ServicesGetter
-	network       *net.IPNet
-	alloc         rangeallocation.RangeRegistry
-	leaks         map[string]int // counter per leaked IP
-	recorder      record.EventRecorder
+
+	network          *net.IPNet
+	alloc            rangeallocation.RangeRegistry
+	secondaryNetwork *net.IPNet
```

Comment:
This is OK, and I don't see anythign egregiously wrong in here, but it seems complicated and fiddly.  2 thoughts:

Would it be easier to just run 2 repair loops, 1 for each family?  That way the delta here becomes "if the IP is not from my family, ignore it" ?

Would it be easier just loop over a list of {range, registry} ignoring family, and only complaining if we get to the end of the list and find no matching CIDR?

Reply:
I would not say easier. testing two distinct `repairers` in a dualstack would be a lot more fiddly. At the runtime each repairer will have to enumerate the entire data set, knowing that it may ignore the entire data set (user choose all services to be one family). The caller (that owns the ref to repairer, now has to handle one type of failure across all types of families supported by the cluster). If we created two repairer will force the caller to handle distinct error cases for each repairer.